### PR TITLE
feat(core): encode #4 ownership boundary and dependency strategy semantics

### PR DESCRIPTION
**Pull request by MangMax**
## Related Issue

Refs #4 — **not** `Closes`. Two acceptance criteria are not fully satisfied; see *Remaining blockers* below.

## What changed

Projects the Issue #4 decision into `@forguncy-react-workspace/core` so it is executable rather than prose. The design deliberation is recorded on [#4](https://github.com/Mang-X/forguncy-react-workspace/issues/4#issuecomment-5710235526); no `specs/` or `plans/` tree was added.

| File | Change |
| --- | --- |
| `packages/core/src/ownership.ts` | The invariant as a concern table (`ownerOf`, `assertOwnedBy`, `concernsOwnedBy`, `OwnershipViolationError`) |
| `packages/core/src/strategy.ts` | `DEPENDENCY_STRATEGY_SEMANTICS` for `host`/`inline`/`extension`/`replace`; decision records; shape vs runtime-verification validation |
| `packages/core/src/rejection.ts` | `architectural` vs `technical` as a discriminated union with opposite response policies |
| `packages/core/src/platform-conflicts.ts` | Role-first platform-conflict assessment + `assessDependencyRole`, with role mismatch kept distinct from ownership conflict |
| `packages/core/src/governance.ts` | The architecture decision back-reference, exported for later Specs |
| `packages/core/src/index.ts` | Barrel export; keeps `DependencyStrategy` / `DependencyDecision` names |
| `AGENTS.md` | Adds only durable boundary rules (role-sensitivity; architectural rejection vs technical bundling failure). **No Issue links and no run state** |
| `.github/ISSUE_TEMPLATE/spec.yml` | New `governing_spec` field, so a Spec declares the architecture decision it follows |
| `.github/workflows/ci.yml` | Adds `vp test --run` and `vp run typecheck` so tests and type contracts are merge gates |
| `tsconfig.typecheck.json` | Package-scoped type-check project (see *Known scope* below) |
| `package.json` | Pins `typescript`, `@types/node`, `vitest` as devDependencies; adds the `typecheck` script |
| `packages/core/src/*.test.ts` | 67 tests across 5 files |

### Contract changes worth reviewing

1. `ReplaceDependencyDecision.reason: string` → `rejection: DependencyRejection`. A free-form string cannot distinguish an architectural rejection from a bundling failure.
2. `DependencyRejection` is a **discriminated union**, not `{ kind; code }`. The type makes a contradictory record impossible to write rather than merely discouraged.
3. `validateDependencyDecision` / `assertDependencyDecision` now require a `DependencyVerificationEvidence` argument. Shape-only callers use `validateDependencyDecisionShape`.
4. `PlatformConflictAssessment` gained `status: "role-mismatch"`, so a wrong-tool pairing is no longer reported as a platform conflict.

### Key design decisions (details in the Issue comment)

- A strategy is a per-(package, cell target) deployment decision, not a package property — hence no compatibility database.
- **Ownership is decided before package classification.** Any package, known or not, requested for an application-owned role is a platform conflict; only cell-local roles consult the package rules. `zustand` for one cell's local state is allowed, the same package as the cross-cell business store is a conflict.
- **Role mismatch ≠ ownership conflict.** A package that has *some* legitimate cell-local roles (React Query) requested for a different cell-local role is a role mismatch. A package with *no* legitimate in-cell use at all (routers, auth frameworks) stays a genuine architectural conflict. Both branches are derived from the rule data, not hardcoded per package.
- Every strategy carries `realRuntimeRequired: true`, and that claim is enforced uniformly instead of only for `extension`.

### Where the decision citation lives

`AGENTS.md` states durable rules only and deliberately carries no URL and no Issue number. The back-reference lives where it can be enforced: the Spec issue template's `governing_spec` field, the exported `OWNERSHIP_AND_DEPENDENCY_DECISION` constant, and a governance test that **fails if `AGENTS.md` ever regains a URL or an Issue reference**. `citesDecision` accepts three boundary-checked forms — `owner/repo#4`, `#4`, and `.../issues/4` — so it agrees with `formatDecisionReference()` while still rejecting `#40`/`#42`, another repository's Issue #4, and a longer path that merely ends in the repository name.

## Contract / spec compliance

- Governing Spec Issue(s): #4
- [x] The implementation stays within the Issue's stated non-goals — no bundler behaviour, no package compatibility database, no runtime API facade.

## Validation

All three commands run in the `check` CI job:

| Command | Result |
| --- | --- |
| `vp check` | 0 errors, 3 warnings — all pre-existing, inside the vendored `forguncy-frontend-library` assets; identical to the pre-change baseline |
| `vp test --run` | 5 files, **67 tests passed** |
| `vp run typecheck` | exit 0 |

The type-level regression tests are load-bearing, not decorative: CI runs a typecheck, and I confirmed the gate is live by checking that an unused `@ts-expect-error` fails with TS2578.

- [ ] Real Forguncy runtime validation — **not executed**. Nothing in this change runs inside Forguncy; all four strategies are declared `realRuntimeRequired: true` for that reason. Target facts still come from #5.

### Known scope of the typecheck

`tsconfig.typecheck.json` covers `packages/**` only. The repository-level `tsconfig.json` still includes `examples/**`, which currently fails because React types are not installed — pre-existing repository bootstrap work, not introduced here. Repo-wide type checking is therefore still open, but the contract this PR adds is now gated.

## Dependency strategy changes

- [x] None — this PR defines the semantics; it does not select a strategy for any package.

## Remaining blockers

1. **AC3 is not satisfied — "The dependency Skill can explain an architectural rejection separately from a technical bundling failure."** `core` now supplies the typed vocabulary and report wording the Skill needs, but the dependency-selection Skill is #18 and does not exist. The only installed Skill is `forguncy-frontend-library`, which is a *packaging* Skill pinned to `MangMax/forguncy-frontend-library` by `skills-lock.json`; editing it in this repository would drift the pinned hash and still would not be the dependency-selection Skill. Blocked by #18 (and the upstream repository that hosts it).
2. **AC2 is only partially satisfiable — "Later Specs reference this Issue when deciding ownership."** The `governing_spec` template field, the exported decision constant and the governance test give later Specs a mechanical way to cite #4, but the criterion cannot be closed until those Specs are authored (#8, #6, #27, #12).
3. **No real Forguncy runtime validation was executed** (see above). #25 is the gate.
4. Repository-wide type checking (including `examples/**`) is still not wired up. This PR gates `packages/**`; the rest remains repository bootstrap work.

Nothing here is self-merged; awaiting review.

---

Notes:

- The branch was created as `spec-4-ownership-and-strategy-semantics` locally and pushed as `spec/4-ownership-and-strategy-semantics`, because the sandboxed shell in this environment silently drops creation of new nested ref directories under `refs/heads/`. The pushed ref name is the intended one.
- No lockfile is committed, consistent with the repository's current state. The new devDependencies are pinned to exact versions for that reason.

### CODE REVIEW

**Comment by MangMax** [comment]
CI confirmation for the validation section above:

- `check` (GitHub Actions, `vp check`) — **pass**, 14s, run [35191454730](https://github.com/Mang-X/forguncy-react-workspace/actions/runs/35191454730/job/105104722990)
- PR state: `OPEN`, `MERGEABLE`

This is the lint/format gate only. It does not cover the type-level contract, and it is not Forguncy runtime validation — both are listed as remaining blockers above.

**Comment by MangMax** [comment]
Follow-up commit `0aaa436` — review feedback applied.

**`AGENTS.md` no longer carries a decision link or any run state.** It now states durable rules only; the governing Spec link and Issue number were removed, and a rule about naming the governing Spec Issue was kept generic.

**Where the citation requirement moved.** It now lives where it can actually be enforced — the Spec issue template keeps its `governing_spec` field, and the exported `OWNERSHIP_AND_DEPENDENCY_DECISION` constant stays in `core`. The governance test was inverted accordingly: instead of asserting that `AGENTS.md` cites the decision, it now asserts that `AGENTS.md` contains **no URL and no Issue reference**, so the former cannot silently come back.

Two further test fixes in the same commit:

- the `specs/`/`plans/` assertion now actually checks that neither directory exists (it previously only checked the PR template wording);
- the PR-template wording check was split into its own case.

Re-validated locally after the change: `vp test --run` → 5 files / **43 tests passed**; `vp check` → 0 errors, 3 warnings (unchanged pre-existing ones in the vendored Skill assets); `tsc --noEmit --strict` on the core entry point → exit 0.

File: packages/core/src/rejection.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  | "dynamic-module-loading"
+  | "global-namespace-collision"
+  | "cell-code-budget-exceeded"
+  | "runtime-api-unavailable";
+
+export type DependencyRejectionCode = ArchitecturalRejectionCode | TechnicalRejectionCode;
+
+export interface DependencyRejection {
+  readonly kind: DependencyRejectionKind;
+  readonly code: DependencyRejectionCode;
```
Comment (MangMax) [inline packages/core/src/rejection.ts]: Blocking: `kind` and `code` are independent unions here, so TypeScript accepts contradictory records such as `{ kind: "architectural", code: "amd-umd-branch-mismatch", ... }`. That defeats the main invariant this module is meant to encode: downstream logic will route by `kind` while code-family helpers route by `code`. Please make `DependencyRejection` a discriminated union that pairs `architectural` with `ArchitecturalRejectionCode` and `technical` with `TechnicalRejectionCode`, and add a regression/type test for the invalid cross-pairing.

File: packages/core/src/platform-conflicts.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+ * package has no legitimate cell-local role at all.
+ */
+export function assessDependencyRole(input: {
+  readonly packageName: string;
+  readonly role: DependencyRole;
+}): PlatformConflictAssessment {
+  const { packageName, role } = input;
+  const rule = findPlatformConflictRule(packageName);
+
+  if (!rule) {
```
Comment (MangMax) [inline packages/core/src/platform-conflicts.ts:270]: Blocking: this returns `unclassified` before considering whether `role` is application-owned. As a result, any package not present in `PLATFORM_CONFLICT_RULES` can be requested as `application-navigation`, `application-state`, `application-auth`, or `business-data-source` and bypass the ownership boundary entirely. The inverse can also misclassify a known package used for an unrelated cell-local role. Please make ownership/role the first decision, then use package-specific rules to refine legitimate cell-local cases. Add tests for unknown-package + each application-owned role.

File: packages/core/src/governance.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * Tokens a governance document must contain to count as citing the decision.
+ * Kept as data so the governance test and any future lint share one definition.
+ */
+export const DECISION_CITATION_TOKENS: readonly string[] = [
+  OWNERSHIP_AND_DEPENDENCY_DECISION_REFERENCE,
+  OWNERSHIP_AND_DEPENDENCY_DECISION.url,
+];
+
+export function citesDecision(text: string): boolean {
+  return DECISION_CITATION_TOKENS.some(token => text.includes(token));
```
Comment (MangMax) [inline packages/core/src/governance.ts]: `includes("#4")` also matches `#40`, `#42`, etc., so `citesDecision("Governing architecture Spec Issue(s): #42")` returns true. Since this helper is intended for governance checks/lint, please use boundary-aware matching or parse issue references explicitly, and add negative tests for `#40`/`#42` and `/issues/40`.

File: packages/core/src/strategy.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+      problems.push(`Extension decision for "${decision.packageName}" must name both the library id and the global.`);
+    }
+    if (options.realRuntimeValidated !== true) {
+      problems.push(
+        `Extension decision for "${decision.packageName}" depends on a host capability that can only be verified in a real Forguncy project.`,
+      );
+    }
+  }
+
+  if (decision.strategy === "host" && decision.globalName.trim().length === 0) {
```
Comment (MangMax) [inline packages/core/src/strategy.ts:257]: API consistency: `host` semantics say `realRuntimeRequired: true` and require the declared global/module identity to be verified in a real Forguncy runtime, but this validator accepts any non-empty `globalName`; `realRuntimeValidated` is enforced only for `extension`. Either make this function explicitly shape-only (and name/document it that way), or apply runtime-evidence validation consistently to every strategy whose semantics mark it required.

**Review by MangMax — COMMENTED** [review]
The overall direction matches #4, but I found two contract-level issues that can undermine the ownership/rejection semantics, plus two merge-gate gaps.

Blocking before merge:
1. `DependencyRejection` does not couple `kind` to the matching code family, so contradictory records are valid TypeScript.
2. `assessDependencyRole()` checks package classification before ownership of the requested role, so an unknown package can fill an application-owned role and fall through as `unclassified`.

Also please fix the `#4` citation substring false-positive and make `vp test --run` part of CI; the current PR check only runs `vp check`, so the 43 new contract tests are not actually a merge gate.

One additional API consistency point is noted inline: `host` declares real-runtime validation as required, but `validateDependencyDecision()` only enforces runtime evidence for `extension`.

I would hold the merge until the two blocking contract issues are fixed.

File: packages/core/src/rejection.ts (hunk 1 above)
Comment (MangMax) [reply packages/core/src/rejection.ts]: Fixed in `4d8a7e1`. `DependencyRejection` is now a discriminated union:

```ts
export interface ArchitecturalDependencyRejection { kind: "architectural"; code: ArchitecturalRejectionCode; ... }
export interface TechnicalDependencyRejection { kind: "technical"; code: TechnicalRejectionCode; ... }
export type DependencyRejection = ArchitecturalDependencyRejection | TechnicalDependencyRejection;
```

`isArchitecturalRejection` / `isTechnicalRejection` are real type predicates now, and `groupRejectionsByKind` returns typed buckets, so a consumer that routes by `kind` cannot receive a code from the other family either.

Regression test added in `rejection.test.ts`: a `@ts-expect-error` pair for both cross-pairings, plus a narrowing test that asserts the code family follows the narrowed kind. The type test is not decorative — CI now runs a typecheck, and I verified the gate is live by asserting that an unused `@ts-expect-error` fails with TS2578.

File: packages/core/src/platform-conflicts.ts (hunk 2 above)
Comment (MangMax) [reply packages/core/src/platform-conflicts.ts:270]: Fixed in `4d8a7e1`. Ownership is now the first decision:

1. If the requested role is application-owned, the request is a platform conflict regardless of package. A rule that duplicates that exact concern supplies the precise code and guidance; otherwise the rejection is `ownership-boundary-violation` with the concern derived from the role.
2. Only for cell-local roles do the package rules decide anything, via `allowedCellLocalRoles`.

Tests added for unknown packages (`es-toolkit`, `my-in-house-router`, `@acme/app-store`, `some-unknown-auth-sdk`) across all four application-owned roles, asserting `kind: "architectural"` and `code: "ownership-boundary-violation"` with `rule` absent. Also covered: an unknown package stays `unclassified` for each cell-local role; a known package asked to fill an *unrelated* application role stays generic while still reporting the matched rule in `evidence`; and no rule may list an application-owned role in `allowedCellLocalRoles`.

File: packages/core/src/governance.ts (hunk 3 above)
Comment (MangMax) [reply packages/core/src/governance.ts]: Fixed in `4d8a7e1`. `DECISION_CITATION_TOKENS` (substring `includes`) is replaced by `DECISION_CITATION_PATTERNS`, matched boundary-aware and derived from the decision constant so there is still one source of truth:

- `(?:^|[^0-9A-Za-z_#])#4(?![0-9A-Za-z_])`
- `/issues/4(?![0-9])`

Negative tests added for `#40`, `#42`, `#44`/`#48`, `#4a`, and `/issues/40` / `/issues/42`, alongside positive cases for `#4`, `(#4)` and the URL.

File: packages/core/src/strategy.ts (hunk 4 above)
Comment (MangMax) [reply packages/core/src/strategy.ts:257]: Fixed in `4d8a7e1` by taking the second option: runtime-evidence validation is now applied uniformly.

The single function was split so the two questions cannot be confused:

- `validateDependencyDecisionShape(decision)` — static self-consistency only, documented as shape-only. It reports no problem for a decision that is merely unverified.
- `validateDependencyVerification(decision, evidence)` — applies the `realRuntimeRequired` claim from each strategy's own semantics, so `host` is no longer exempt.
- `validateDependencyDecision(decision, evidence)` — both, with `evidence` a **required** argument so the runtime claim cannot be omitted implicitly.
- `assertDependencyDecision(decision, evidence)` — throws on both.

Tests assert that every strategy in `DEPENDENCY_STRATEGIES` is gated while `requiresRealRuntimeValidation` holds, that shape problems never mention verification, and that `host` now reports `"react" cannot be reported as verified` until the flag is supplied.

File: packages/core/src/platform-conflicts.ts [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+ * the rule does not permit.
+ */
+function rejectionForRole(rule: PlatformConflictRule, role: DependencyRole, packageName: string): DependencyRejection {
+  return {
+    kind: "architectural",
+    code: rule.code,
+    summary:
+      rule.allowedCellLocalRoles.length === 0
+        ? `"${packageName}" has no legitimate cell-local role: "${role}" is the Forguncy-owned concern "${rule.concern}".`
+        : `"${packageName}" was requested for "${role}", which duplicates the Forguncy-owned concern "${rule.concern}".`,
```
Comment (MangMax) [inline packages/core/src/platform-conflicts.ts]: Follow-up: the ownership-first bypass is fixed, but the inverse false-positive still exists here. For example, `assessDependencyRole({ packageName: "@tanstack/react-query", role: "cell-local-state" })` reaches this branch because the package rule only allows `cell-local-data-access`; it then returns an **architectural** `duplicate-business-data-source` rejection and says the cell-local role duplicates the Forguncy-owned data-source concern. But `cell-local-state` is explicitly a cell-local role, so the requested role itself does not cross the ownership boundary. This conflates "this package is not a sensible implementation for that local role" with an architectural platform conflict. Please distinguish packages that truly have no legitimate in-cell use at all (router/auth) from a known package used for an unrelated but still cell-local role; the latter should stay unclassified / role-mismatch rather than become an ownership rejection. Add a regression case for React Query + `cell-local-state` (and/or `cell-local-ui`).

File: packages/core/src/governance.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ * Patterns a governance document must match to count as citing the decision.
+ *
+ * Deliberately boundary-aware rather than substring-based: `#4` is a prefix of
+ * `#40`, `#42` and so on, so `text.includes("#4")` would report a citation of a
+ * completely different Issue as a citation of this one. Exported as data so the
+ * governance test and any future lint share one definition.
+ */
+export const DECISION_CITATION_PATTERNS: readonly RegExp[] = [
+  // `#4` as a whole reference, not the head of a longer number or identifier.
+  new RegExp(`(?:^|[^0-9A-Za-z_#])#${OWNERSHIP_AND_DEPENDENCY_DECISION.issue}(?![0-9A-Za-z_])`),
```
Comment (MangMax) [inline packages/core/src/governance.ts]: The `#40` false-positive is fixed, but this boundary is now too strict for the canonical repo-qualified short form. `formatDecisionReference()` emits `Mang-X/forguncy-react-workspace#4 — ...`; immediately before `#` is the alphanumeric `e`, so this regex does not match and `citesDecision(formatDecisionReference())` returns `false`. Since both helpers are exported as the canonical provenance API, they should agree. Please recognize the exact `${repository}#${issue}` form (without reintroducing `#40` prefix matching) and add `expect(citesDecision(formatDecisionReference())).toBe(true)`.

**Review by MangMax — COMMENTED** [review]
Re-review of `4d8a7e1`: the original discriminated-union issue, ownership-first bypass, runtime-verification consistency, and CI merge-gate gap are fixed. Latest CI is green with `vp check`, `vp test --run` (59 tests), and `vp run typecheck`.

I found two remaining edge cases before I would consider this clean:

1. The cell-local branch can still turn a package/role mismatch into an architectural ownership conflict. In particular, `@tanstack/react-query` + `cell-local-state` reports `duplicate-business-data-source`, even though `cell-local-state` is explicitly not application-owned. This is the inverse false-positive from the previous review.
2. The new boundary-aware citation matcher rejects the canonical repo-qualified short reference produced by `formatDecisionReference()` (`Mang-X/forguncy-react-workspace#4`).

Both are commented inline with concrete regression cases.

File: packages/core/src/platform-conflicts.ts (hunk 5 above)
Comment (MangMax) [reply packages/core/src/platform-conflicts.ts]: Confirmed and fixed in `12417cf`. I reproduced it first: `@tanstack/react-query` + `cell-local-state` returned `platform-conflict` / `duplicate-business-data-source`, and so did `+ cell-local-ui`.

The fix uses the distinction you drew, and it is derivable from the rule data rather than hardcoded:

- `allowedCellLocalRoles` is **empty** → the package has no legitimate in-cell use at all (routers, auth frameworks), so any in-cell request stays a real architectural conflict. `react-router-dom` and `@auth0/auth0-react` + `cell-local-ui` still report `application-router-conflict` / `auth-framework-conflict`.
- `allowedCellLocalRoles` is **non-empty** → the package *is* usable in a cell, just for those roles. A different cell-local role now returns the new `status: "role-mismatch"` (with `allowedCellLocalRoles` on the assessment) instead of an architectural rejection, and `isPlatformConflict` is `false` for it.

Regression tests added for `@tanstack/react-query` + `cell-local-state` **and** `+ cell-local-ui`, the preserved router/auth cases, and a table-driven check that every rule lands on the correct side of the line.

File: packages/core/src/governance.ts (hunk 6 above)
Comment (MangMax) [reply packages/core/src/governance.ts]: Partly confirmed, and fixed in `12417cf` — but the reported symptom does not reproduce as stated, so the fix is in a different place than described.

**What I measured** on `4d8a7e1` before changing anything:

```
FORMATTED: "Mang-X/forguncy-react-workspace#4 — Spec: ... (https://github.com/Mang-X/forguncy-react-workspace/issues/4)"
citesDecision(formatDecisionReference()) = true
citesDecision("Mang-X/forguncy-react-workspace#4")       = false
```

`citesDecision(formatDecisionReference())` already returned **`true`** — that string also embeds the full URL, which the `/issues/4` pattern matches. So the assertion you asked for was already satisfied and would not have failed.

**The real gap is exactly what you identified underneath it:** the bare repo-qualified short form was not recognised, even though `formatDecisionReference` presents `owner/repo#4` as the canonical reference. That is fixed: a third pattern now matches `${repository}#${issue}` anchored on the exact repository, so the `#40` false positive stays fixed.

Tests added: `citesDecision(formatDecisionReference()) === true`, `citesDecision(OWNERSHIP_AND_DEPENDENCY_DECISION_QUALIFIED_REFERENCE) === true`, an inline-context case, plus negatives for `Mang-X/forguncy-react-workspace#40`/`#42`, `other-org/other-repo#4` and `Mang-X/other-repo#4`. I also exported the qualified form as `OWNERSHIP_AND_DEPENDENCY_DECISION_QUALIFIED_REFERENCE` so the two helpers share one source of truth.

File: packages/core/src/governance.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+export const DECISION_CITATION_PATTERNS: readonly RegExp[] = [
+  // 1. owner/repo#4 — anchored on the exact repository so this cannot degrade
+  //    into prefix matching.
+  new RegExp(
+    `${escapeForRegExp(OWNERSHIP_AND_DEPENDENCY_DECISION.repository)}#${OWNERSHIP_AND_DEPENDENCY_DECISION.issue}(?![0-9A-Za-z_])`,
+  ),
+  // 2. `#4` as a whole reference, not the head of a longer number or identifier.
+  new RegExp(`(?:^|[^0-9A-Za-z_#])#${OWNERSHIP_AND_DEPENDENCY_DECISION.issue}(?![0-9A-Za-z_])`),
+  // 3. `.../issues/4`, again not the head of `/issues/40`.
+  new RegExp(`/issues/${OWNERSHIP_AND_DEPENDENCY_DECISION.issue}(?![0-9])`),
```
Comment (MangMax) [inline packages/core/src/governance.ts]: Remaining provenance bug: this URL pattern is still repository-agnostic. `citesDecision("https://github.com/other-org/other-repo/issues/4")` returns `true`, so a citation to some other repository's Issue #4 satisfies the check for `Mang-X/forguncy-react-workspace#4`. That undermines the mechanical AC2 back-reference this helper is meant to enforce. Please match the canonical decision URL (or at least the exact escaped repository path + `/issues/${issue}`) and add negatives for `other-org/other-repo/issues/4` / `Mang-X/other-repo/issues/4`. The repo-qualified pattern could likewise use a left boundary if you want its “exact repository” claim to be literal.

File: packages/core/src/platform-conflicts.ts [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+ * package has no legitimate in-cell use at all. A cell-local role mismatch does
+ * not reach here — see `PlatformConflictRoleMismatch`.
+ */
+function rejectionForRole(rule: PlatformConflictRule, role: DependencyRole, packageName: string): DependencyRejection {
+  return {
+    kind: "architectural",
+    code: rule.code,
+    summary:
+      rule.allowedCellLocalRoles.length === 0
+        ? `"${packageName}" has no legitimate cell-local role: "${role}" is the Forguncy-owned concern "${rule.concern}".`
```
Comment (MangMax) [inline packages/core/src/platform-conflicts.ts]: The classification is fixed, but this rejection text is now semantically wrong for the preserved router/auth cases. For `react-router-dom + cell-local-ui`, it says `"cell-local-ui" is the Forguncy-owned concern "application-navigation"`; for an auth framework it analogously equates the local UI role with `permissions`. The *package* necessarily implements a host-owned concern and has no legitimate in-cell role, but the requested cell-local role itself is not host-owned. Please phrase this as e.g. `"react-router-dom" implements the Forguncy-owned concern "application-navigation" and has no legitimate cell-local role; it cannot be used for "cell-local-ui"`, and add a regression assertion on the summary so the reporting layer cannot reintroduce this conflation.

**Review by MangMax — COMMENTED** [review]
Re-review of `12417cf`: the two findings from the previous pass are fixed correctly. `@tanstack/react-query` in an unrelated cell-local role now becomes `role-mismatch` rather than an architectural rejection, the repo-qualified `owner/repo#4` form is recognized, and the latest CI run passes `vp check`, tests, and package typecheck.

I found two remaining accuracy issues below. Neither reopens the earlier ownership/type-contract bugs, but both affect the reliability of the executable reporting/governance layer, so I would fix them before merging. I did not find other new regressions in the follow-up commit.

File: packages/core/src/governance.ts (hunk 7 above)
Comment (MangMax) [reply packages/core/src/governance.ts]: Confirmed and fixed in `d31a0b1`. Reproduced first — all four of these returned `true` before the fix:

```
https://github.com/other-org/other-repo/issues/4          -> true   (wrong)
https://github.com/Mang-X/other-repo/issues/4             -> true   (wrong)
https://github.com/Mang-X/forguncy-react-workspace/issues/4 -> true
foo/Mang-X/forguncy-react-workspace#4                     -> true   (wrong)
```

Both points addressed:

- The URL pattern is now anchored on the exact escaped repository path: `${repository}/issues/${issue}(?![0-9])`. Another repository's Issue #4 can no longer satisfy this decision's check.
- The repo-qualified pattern took the left boundary its "exact repository" claim implied: `(?:^|[^0-9A-Za-z_./-])`, so a longer path merely ending in the repository name is rejected, while `^`, whitespace, `(` and backticks still work.

Tests: negatives for `https://github.com/other-org/other-repo/issues/4`, `https://github.com/Mang-X/other-repo/issues/4`, `other-org/other-repo#4`, `Mang-X/other-repo#4`, `foo/Mang-X/forguncy-react-workspace#4`, `not-Mang-X/forguncy-react-workspace#4`; positive for `OWNERSHIP_AND_DEPENDENCY_DECISION.url` itself, so the canonical URL stays provably recognized.

File: packages/core/src/platform-conflicts.ts (hunk 8 above)
Comment (MangMax) [reply packages/core/src/platform-conflicts.ts]: Confirmed and fixed in `d31a0b1`. Reproduced first — the summary read exactly as you quoted:

```
"react-router-dom" has no legitimate cell-local role: "cell-local-ui" is the Forguncy-owned concern "application-navigation".
"@auth0/auth0-react" has no legitimate cell-local role: "cell-local-ui" is the Forguncy-owned concern "permissions".
```

Reworded along the lines you suggested:

```
"react-router-dom" implements the Forguncy-owned concern "application-navigation" and has no legitimate cell-local role, so it cannot be used for "cell-local-ui".
```

The two summary branches are now explicitly documented as different cases, because the distinction is exactly what was being lost:

- empty `allowedCellLocalRoles` → the **package** implements a host-owned concern and has no in-cell use;
- application-owned role → the **role** itself is the host-owned concern, so the original phrasing is correct there and is kept.

Regression assertions added on the summary text for both branches: the router/auth summaries must contain `implements the Forguncy-owned concern "<concern>"` and must **not** match `/"cell-local-ui" is the Forguncy-owned concern/`, while the `zustand + application-state` summary must still say the role *is* the concern.

**Review by MangMax — COMMENTED** [review]
Re-reviewed at `d31a0b1`.

The two findings from the previous round are resolved:

- Decision URL provenance is now tied to `Mang-X/forguncy-react-workspace`; another repository's Issue #4 no longer satisfies `citesDecision`, and the repo-qualified form also has the missing left boundary. The new negative/positive cases cover the regression.
- Router/auth rejections for cell-local roles now blame the package for implementing the Forguncy-owned concern instead of falsely describing the requested cell-local role as host-owned. The regression assertions preserve the distinction from the application-owned-role branch.

I also re-checked the surrounding role-first/role-mismatch logic and did not find a new contract or ownership regression in this follow-up commit.

Latest CI is green with all three gates (`vp check`, `vp test --run` — 67 tests, and `vp run typecheck`).

No additional code-level blocking findings from this review. The remaining items listed in the PR body (later Specs/Skill/runtime validation/repo-wide example typecheck) are explicitly tracked outside the implementation delivered here, so keeping this PR as `Refs #4` rather than `Closes #4` remains appropriate.
